question-answering-using-openai/main.bal

```diff
@@ -0,0 +1,211 @@
+import ballerina/http;
+import ballerinax/googleapis.sheets;
+import ballerinax/openai;
+import ballerina/lang.'float as langFloat;
+import ballerina/regex;
+import ballerina/log;
+import ballerina/os;
+
+
+// Google Sheets API client configuration parameters
+final string gsheetsClientID = os:getEnv("GSHEETS_CLIENT_ID");
+final string gsheetsClientSecret = os:getEnv("GSHEETS_CLIENT_SECRET");
+final string gsheetsRefreshToken = os:getEnv("GSHEETS_REFRESH_TOKEN");
```

Comment:
Use configurables?

Reply:
could you elaborate what you mean by token based approach?

Reply:
```ballerina
configurable string sheetsAccessToken = ?;
sheets:Client gsheets = check new ({auth: {token: sheetsAccessToken}});
```

Reply:
noted and addressed in [a3d99a2](https://github.com/ballerina-guides/ai-samples/pull/2/commits/a3d99a29fc85ec81cafb7f92df253d5cbe4b6449)